operator/helper/multiline.go

```diff
@@ -19,62 +19,125 @@ import (
 	"bytes"
 	"fmt"
 	"regexp"
+	"time"
 
 	"golang.org/x/text/encoding"
-
-	"github.com/open-telemetry/opentelemetry-log-collection/operator"
 )
 
+type ForceFlush struct {
+	Force     bool
+	LastFlush time.Time
+}
+
+func NewForceFlush() *ForceFlush {
+	return &ForceFlush{
+		Force:     false,
+		LastFlush: time.Now(),
+	}
+}
+
+type Multiline struct {
+	SplitFunc   bufio.SplitFunc
+	force       *ForceFlush
+	forcePeriod time.Duration
+
+	// lastFlush > force.LastFlush => we can force flush if no logs are incoming for forcePeriod
+	// lastFlush = force.LastFlush => last flush was forced, so we do cannot force, we can update lastFlush
+	// lastFlush < force.LastFlush =>we just forced flush, set lastFlush to force.LastFlush
+	lastFlush time.Time
+}
+
+// Flushed update lastFlush with current timestamp
+func (m *Multiline) Flushed() {
+	if m.lastFlush.Sub(m.force.LastFlush) < 0 {
+		m.lastFlush = m.force.LastFlush
+	} else {
+		m.lastFlush = time.Now()
+	}
+}
+
+// CheckAndFlush returns true if data is going to be force flushed
```

Comment:
It doesn't actually return anything. It sets the flag.

Reply:
fixed